File: src/industry.ts
export interface IndustryResearchRequest {
  question: string;
  industry?: string;
  region?: string;
  timeRange?: string;
  companies?: string[];
}

const DEFAULT_DIMENSIONS = [
  'market size and growth',
  'competitive landscape',
  'technology and product trends',
  'regulation and policy',
  'key risks and uncertainties',
];

export function buildIndustryPrompt(request: IndustryResearchRequest) {
  const constraints = [
    request.industry && `Industry: ${request.industry}`,
    request.region && `Region: ${request.region}`,
    request.timeRange && `Time range: ${request.timeRange}`,
    request.companies?.length && `Companies: ${request.companies.join(', ')}`,
  ].filter(Boolean);

  return [
    request.question,
    constraints.length
      ? `Research constraints:\n${constraints.join('\n')}`
      : '',
    `Unless the question clearly excludes them, cover:\n${DEFAULT_DIMENSIONS.map(item => `- ${item}`).join('\n')}`,
    'Prefer primary, regulator, official, academic, and reputable media sources. Preserve conflicting evidence.',
  ]
    .filter(Boolean)
    .join('\n\n');
}

File: evidence_research/industry.py
from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(frozen=True)
class IndustryRequest:
    question: str
    industry: str | None = None
    region: str | None = None
    time_range: str | None = None
    companies: list[str] = field(default_factory=list)


DEFAULT_DIMENSIONS = (
    "market size and growth",
    "competitive landscape",
    "technology and product trends",
    "regulation and policy",
    "key risks and uncertainties",
)


def build_industry_prompt(request: IndustryRequest) -> str:
    constraints = [
        f"Industry: {request.industry}" if request.industry else None,
        f"Region: {request.region}" if request.region else None,
        f"Time range: {request.time_range}" if request.time_range else None,
        f"Companies: {', '.join(request.companies)}" if request.companies else None,
    ]
    constraints = [item for item in constraints if item]
    parts = [request.question]
    if constraints:
        parts.append("Research constraints:\n" + "\n".join(constraints))
    parts.append("Unless excluded, cover:\n" + "\n".join(f"- {item}" for item in DEFAULT_DIMENSIONS))
    parts.append(
        "Prefer primary, regulator, official, academic, and reputable media sources. "
        "Preserve conflicting evidence."
    )
    return "\n\n".join(parts)
